web/components/FcPaneMap.vue

```diff
@@ -234,30 +234,87 @@ export default {
       };
     },
     locationsMarkersGeoJson() {
-      const features = this.locationsForMode.map((location, i) => {
+      const waypoints = this.locationsSelectionForMode.locations;
+
+      let waypointIndex = 0;
+      let waypoint = waypoints[waypointIndex];
+      const features = [];
+      this.locationsForMode.forEach((location) => {
         const {
           geom,
           lat,
           lng,
-          ...properties
+          ...propertiesRest
         } = location;
+
         const geometry = {
           type: 'Point',
           coordinates: [lng, lat],
         };
-        const { multi } = this.locationMode;
-        properties.multi = multi;
-        if (multi) {
-          properties.locationIndex = i;
-          properties.selected = i === this.locationEditIndex;
+
+        /*
+         * It is possible for the current location to match multiple consecutive waypoints.
+         * We could forbid selecting the same location multiple times, but that would introduce
+         * a lot of validation complexity in both frontend and backend.  It would also make it
+         * impossible to select a corridor that loops back on itself.
+         */
+        const waypointIndices = [];
+        while (waypointIndex < waypoints.length
```

Comment:
can we set `const n = waypoints.length` above and use that here?  (this helps preserve the common `k` of `n` naming convention from various branches of mathematics, which would probably help make it clearer that `waypointIndices` identifies a subsequence of `waypoints`.  also we should say that explicitly in the comment here.)

Reply:
(I'd also love to get this under test, as it's an outlier among computed properties in terms of its complexity...but that will likely have to wait until we have spare cycles to build up some UI unit tests.)